cloudconfig/cloudinit/network_ubuntu.go

```diff
@@ -323,13 +323,12 @@ if [ ! -f /sbin/ifup ]; then
     exit 0
 fi
 if [ -f /usr/bin/python ]; then
-    python %[1]s.py --interfaces-file %[1]s.tmp
+    python %[1]s.py --interfaces-file %[1]s
 else
-    python3 %[1]s.py --interfaces-file %[1]s.tmp
+    python3 %[1]s.py --interfaces-file %[1]s
```

Comment:
The idea here was to keep the original interfaces file as long as possible (until after ifdown). Maybe modify the python script to accept two parameters - original and 'output' interfaces file?

Reply:
interfaces.py does write to interfaces.tmp, then move interfaces.tmp to interfaces the very last thing.  ifdown used to be called before interfaces.py in 2.3.5.  

For future reference, conversation with wupeka:
the main concern is to wait with anything that's hard to revert for as long as possible, we've seen weird failure modes and IMHO it's impossible to avoid them all, but we could do as much as we can to 1. avoid them 2. make the machine boot if we mess up

Reply:
I thought we wrote out /e/n/i from outside the container after create, but before start and forced it into the container via the LXD api. Did we change it to follow the same pattern as KVM instances and rewrite the /e/n/i as part of bootcmd ?

Reply:
@jameinel it's part of the provisioner container specific items added to bootcmd in the cloud-init user-data juju creates.

Reply:
updated to take an output file and move just before ifup.